core/src/main/scala/org/apache/spark/security/CryptoStreamUtils.scala

```diff
@@ -63,12 +83,40 @@ private[spark] object CryptoStreamUtils extends Logging {
       is: InputStream,
       sparkConf: SparkConf,
       key: Array[Byte]): InputStream = {
-    val properties = toCryptoConf(sparkConf)
     val iv = new Array[Byte](IV_LENGTH_IN_BYTES)
-    is.read(iv, 0, iv.length)
-    val transformationStr = sparkConf.get(IO_CRYPTO_CIPHER_TRANSFORMATION)
-    new CryptoInputStream(transformationStr, properties, is,
-      new SecretKeySpec(key, "AES"), new IvParameterSpec(iv))
+    var read = 0
+    while (read < iv.length) {
```

Comment:
It avoids issues with short reads. It's unlikely to happen but I always write read code like this to be safe.